### PR TITLE
Added shared domain object for data format that is returned by the re…

### PR DESCRIPTION
**Pull request by amobarak**
…sults service

# Description and relevant Jira numbers

New object that will hold the search results, provided from the new search results API.

## Pull Request checklist

- [x] [WIP] tag removed from PR title
- [x] PR has an understandable description

## Git feature branch checklist

- [x] branch name comply with our branching strategy
- [x] git branch contains relevant JIRA ticket number
- [x] branch rebased against the latest develop

## Sign off process checklist

- [x] Code has been tested manually
- [ ] Tested by QA
- [ ] PO's approval
- [ ] Reviewers selected in Github
- [ ] PR link added to JIRA

### CODE REVIEW

File: mes-search-schema/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import { Name, ActivityCode } from "../../../mes-test-schema/categories/B";
+
+/* tslint:disable */
+/**
+ * This file was automatically generated by json-schema-to-typescript.
+ * DO NOT MODIFY IT BY HAND. Instead, modify the source JSONSchema file,
+ * and run json-schema-to-typescript to regenerate this file.
+ */
+
+export interface SearchResultTestCATBSchema {
```
Comment (rstepnowski) [inline mes-search-schema/index.d.ts]: Is this a Cat. B specific thing? There's probably nothing that is cat.B specific, isn't it? 
Are we not planning to use this for searching any category tests? If yes a name change would be appropriate.

File: mes-search-schema/package-lock.json [hunk 2]
```diff
@@ -0,0 +1,340 @@
+{
+  "name": "@dvsa/mes-test-schema",
```
Comment (rstepnowski) [inline mes-search-schema/package-lock.json]: Do we want to name it `mes-test-schema` even though it's in a separate folder?

File: mes-search-schema/package-lock.json [hunk 3]
```diff
@@ -0,0 +1,340 @@
...
+  "version": "1.0.21",
```
Comment (rstepnowski) [inline mes-search-schema/package-lock.json]: If we're going to treat it as a separate thing we probably want to also version it separately

File: mes-search-schema/index.d.ts (hunk 1 above)
Comment (amobarak) [reply mes-search-schema/index.d.ts]: Oh its a typo, will change it to "SearchResultTest"

**Pull request merged**